src/tests/test_rdymesh.c

```diff
@@ -68,6 +68,10 @@ static PetscErrorCode Create2DUnitBoxDM(PetscInt Nx, PetscInt Ny, DM *dm) {
   if (dm_dist) {
     DMDestroy(dm);
     *dm = dm_dist;
+  } else {
+    // Create a global section for later use.
+    PetscSection global_sec;
+    PetscCall(DMGetGlobalSection(*dm, &global_sec));
```

Comment:
I don't believe we need to create a `PetscSection`.

Reply:
The single-process test fails when these lines are omitted, with the following error:

```
$ ./src/tests/test_rdymesh 
[==========] tests: Running 1 test(s).
[ RUN      ] TestRDyMeshCreateFromDM
[0]PETSC ERROR: --------------------- Error Message --------------------------------------------------------------
[0]PETSC ERROR: Invalid argument
[0]PETSC ERROR: DM must have a global section. It will be created automatically by DMGetGlobalSection()
[0]PETSC ERROR: See https://petsc.org/release/faq/ for trouble shooting.
[0]PETSC ERROR: Petsc Development GIT revision: v3.18.1-360-g1249ebefbf6  GIT Date: 2022-11-23 02:34:53 +0000
[0]PETSC ERROR: ./src/tests/test_rdymesh on a debug named trippy by jeff Wed Nov 23 17:47:08 2022
[0]PETSC ERROR: Configure options --CFLAGS=-g --CXXFLAGS=-g --FFLAGS="-g -Wno-unused-function" --download-exodusii --download-fblaslapack --download-hdf5 --download-metis --download-netcdf --download-parmetis --download-pnetcdf --with-clanguage=c --with-debugging=1 --with-cudac=0 --with-shared-libraries=1 --with-zlib PETSC_ARCH=debug
[0]PETSC ERROR: #1 DMGetGlobalOffset_Private() at /home/jeff/projects/pnnl/petsc/include/petsc/private/dmimpl.h:458
[0]PETSC ERROR: #2 DMPlexGetPointGlobal() at /home/jeff/projects/pnnl/petsc/src/dm/impls/plex/plexpoint.c:238
[0]PETSC ERROR: #3 RDyCellsCreateFromDM() at /home/jeff/projects/pnnl/RDycore/src/rdymesh.c:108
[0]PETSC ERROR: #4 RDyMeshCreateFromDM() at /home/jeff/projects/pnnl/RDycore/src/rdymesh.c:496
[  ERROR   ] --- 0 != 62
[   LINE   ] --- /home/jeff/projects/pnnl/RDycore/src/tests/test_rdymesh.c:90: error: Failure!
[  FAILED  ] TestRDyMeshCreateFromDM
[==========] tests: 1 test(s) run.
[  PASSED  ] 0 test(s).
[  FAILED  ] tests: 1 test(s), listed below:
[  FAILED  ] TestRDyMeshCreateFromDM

 1 FAILED TEST(S)
```

Reply:
The following approach of creating/destroying a `Vec` instead of using a `PetscSection` also works. I recall that @knepley had previously provided an example in which such a dummy `Vec` was being used. I don't know which is a better approach (i.e., `Vec` or `PetscSection`). @knepley any thoughts?

```
diff --git a/src/tests/test_rdymesh.c b/src/tests/test_rdymesh.c
index a42fee5..f7e04b6 100644
--- a/src/tests/test_rdymesh.c
+++ b/src/tests/test_rdymesh.c
@@ -68,13 +68,15 @@ static PetscErrorCode Create2DUnitBoxDM(PetscInt Nx, PetscInt Ny, DM *dm) {
   if (dm_dist) {
     DMDestroy(dm);
     *dm = dm_dist;
-  } else {
-    // Create a global section for later use.
-    PetscSection global_sec;
-    PetscCall(DMGetGlobalSection(*dm, &global_sec));
   }
+
   PetscCall(DMViewFromOptions(*dm, NULL, "-dm_view"));

+  // Create a dummy vec
+  Vec dummy;
+  DMCreateGlobalVector(*dm,&dummy);
+  VecDestroy(&dummy);
+
   PetscFunctionReturn(0);
 }
```

Reply:
I think either way we're creating a global section--one is just implicit (and involves the creation and destruction of a `Vec`). The call to `DMPlexGetPointGlobal` requires a global section, if I'm interpreting the error correctly.